nix/modules/holochain.nix

```
@@ -55,14 +55,16 @@
       });
 
       # derivation with the main crates
-      holochain = (craneLib.buildPackage (commonArgs // {
-        CARGO_PROFILE = "release";
-        cargoArtifacts = holochainDepsRelease;
-        src = flake.config.srcCleanedHolochain;
-        doCheck = false;
-      })) // {
-        src.rev = inputs.holochain.rev;
-      };
+      holochain = lib.attrsets.recursiveUpdate
+        (craneLib.buildPackage (commonArgs // {
+          CARGO_PROFILE = "release";
+          cargoArtifacts = holochainDepsRelease;
+          src = flake.config.srcCleanedHolochain;
+          doCheck = false;
+        }))
+        {
+          src.rev = inputs.holochain.rev;
+        };
```

What's the purpose of this? To append attributes to derivations, better use `passthru`, as everything else gets lost as soon as overrides are applied.
```suggestion
      holochain =
        (craneLib.buildPackage (commonArgs // {
          CARGO_PROFILE = "release";
          cargoArtifacts = holochainDepsRelease;
          src = flake.config.srcCleanedHolochain;
          doCheck = false;
          passthru.rev = inputs.holochain.rev;
        }));
```

---

@steveeJ You didn't reply to this - does David's suggestion not apply?

---

FYI the information that's added to the set is read in [`hn-introspect`](https://github.com/holochain/holochain/blob/28d6cb4c56d4c643301c7c5370a5fc37f36798f3/nix/modules/devShells.nix#L9).